buffer/src/main/java/io/netty/buffer/UnpooledUnsafeHeapByteBuf.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Override
+    public int getInt(int index) {
+        checkIndex(index, 4);
+        return _getInt(index);
+    }
+
+    @Override
+    protected int _getInt(int index) {
+        return UnsafeByteBufUtil.getInt(array, index);
+    }
+
+    @Override
+    public long getLong(int index) {
+        checkIndex(index, 8);
+        return _getLong(index);
+    }
+
+    @Override
+    protected long _getLong(int index) {
+        return UnsafeByteBufUtil.getLong(array, index);
+    }
+
+    @Override
+    public ByteBuf setByte(int index, int value) {
+        checkIndex(index);
+        _setByte(index, value);
+        return this;
+    }
+
+    @Override
+    protected void _setByte(int index, int value) {
+        UnsafeByteBufUtil.setByte(array, index, value);
+    }
+
+    @Override
+    public ByteBuf setShort(int index, int value) {
+        checkIndex(index, 2);
+        _setShort(index, value);
+        return this;
+    }
+
+    @Override
+    protected void _setShort(int index, int value) {
+        UnsafeByteBufUtil.setShort(array, index, value);
+    }
+
+    @Override
+    public ByteBuf setMedium(int index, int   value) {
+        checkIndex(index, 3);
+        _setMedium(index, value);
+        return this;
+    }
+
+    @Override
+    protected void _setMedium(int index, int value) {
+        UnsafeByteBufUtil.setMedium(array, index, value);
+    }
+
+    @Override
+    public ByteBuf setInt(int index, int   value) {
+        checkIndex(index, 4);
+        _setInt(index, value);
+        return this;
+    }
+
+    @Override
+    protected void _setInt(int index, int value) {
+        UnsafeByteBufUtil.setInt(array, index, value);
+    }
+
+    @Override
+    public ByteBuf setLong(int index, long  value) {
+        checkIndex(index, 8);
+        _setLong(index, value);
+        return this;
+    }
+
+    @Override
+    protected void _setLong(int index, long value) {
+        UnsafeByteBufUtil.setLong(array, index, value);
+    }
+
+    @Override
+    public ByteBuf setZero(int index, int length) {
+        if (PlatformDependent.javaVersion() >= 7) {
+            // Only do on java7+ as the needed Unsafe call was only added there.
+            checkIndex(index, length);
+            UnsafeByteBufUtil.setZero(array, index, length);
+            return this;
+        }
+        return super.setZero(index, length);
+    }
+
+    @Override
+    public ByteBuf writeZero(int length) {
+        if (PlatformDependent.javaVersion() >= 7) {
+            // Only do on java7+ as the needed Unsafe call was only added there.
+            ensureWritable(length);
+            int wIndex = writerIndex;
+            UnsafeByteBufUtil.setZero(array, wIndex, length);
+            writerIndex = wIndex + length;
+            return this;
+        }
+        return super.writeZero(length);
+    }
+
+    @Override
+    @Deprecated
+    protected SwappedByteBuf newSwappedByteBuf() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)